### ISSUE TITLE
Advanced Build Broken

### ISSUE THREAD

**Issue by elind21** [PapaJoesSoup/BDArmory#692]
When reporting issues please provide the following information:

[See this for more information on reporting issues for KSP Mods](http://forum.kerbalspaceprogram.com/?showtopic=83212)

[See the Original Forum Post (OP) for additional info](http://forum.kerbalspaceprogram.com/?showtopic=155014)

##### ISSUE TYPE
 - Bug Report   (Something is broken)

##### KSP and BDA Version

```
    BDAC 1.3.0 via CKAN
    KSP v1.7.0.2483 with Making History

```

##### OS / ENVIRONMENT
Windows 10 x64
i7-7700K
GTX1070
32GB RAM

##### SUMMARY
In VAB or SPH, opening the advanced filters build menu breaks the category selection menus.

##### STEPS TO REPRODUCE
Clean KSP install (advanced tweakables enabled) with BDA, Module Manager and Physics Range Extender
Open a sandbox game
Enter VAB - categories work as expected
Expand advanced tab (upper left) and select filter by section profile/size
    Expected: categories filter by size and surface
    Actual: categories freeze on whichever tab was open before launching advanced tab

Closing advanced tab does not resolve issue. Exiting and entering VAB/SPH after closing tab resolves issue until advanced tab is opened.

Examining the console, the game throws an exception:
```
[EXC 19:02:48.325] NullReferenceException: Object reference not set to an instance of an object
	KIS.KISAddonConfig.Awake ()
	UnityEngine.GameObject:AddComponent(Type)
	AddonLoader:StartAddon(LoadedAssembly, Type, KSPAddon, Startup)
	AddonLoader:StartAddons(Startup)
	<LoadObjects>c__Iterator1:MoveNext()
	UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
	<CreateDatabase>c__Iterator0:MoveNext()
	UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
	GameDatabase:StartLoad()
	KSP.UI.Screens.DebugToolbar.Screens.Database.<RecompileAssets>c__Iterator0:MoveNext()
	UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
	KSP.UI.Screens.DebugToolbar.Screens.Database.ScreenDatabaseOverview:OnReloadClicked()
	UnityEngine.EventSystems.EventSystem:Update()
```
upon opening the advanced tab.

Rectification attempted:
Remove all parts folders (except weaponManager) from .\GameData\BDArmory\Parts
edit weaponManager.cfg :
    Under // --- standard part parameters --- 
        add a new line:
`bulkheadProfiles = srf`
Reload KSP, repeat reproduction steps - issue no longer occurs.

##### Log Files
Log files from first encounter with issue - other mods installed.

[KSP.log](https://github.com/PapaJoesSoup/BDArmory/files/3165799/KSP.log)

[output_log.txt](https://github.com/PapaJoesSoup/BDArmory/files/3165807/output_log.txt)

**Comment by Gedas-S** [comment]
Thank you for the report.

I suppose we haven't updated BDA parts with `bulkheadProfiles`, and that is what's causing the error.

We'll try to get it fixed, meanwhile you can try using a modulemanager patch to add the property to bda parts as a temporary remedy ( https://forum.kerbalspaceprogram.com/index.php?/topic/106243-170161151145131-interstellar-fuel-switch-ifs-382/&page=39&tab=comments#comment-3531322 ).

Also, the attached `KSP.log` seems incomplete (the copied error is not in it), but I guess it is rather clear what's going on, so no worries.

**Comment by elind21** [comment]
Thanks for the speedy reply - the MM patch fixed the issue.

Regarding the KSP.log - the error is there at [EXC 19:26:11.958]. It is slightly different to the copied one, but the same scenario. The reason it's incomplete is because I force closed KSP with Alt-F4.

I was reading somewhere that there is a method for adding custom bulkheadProfiles. It would be cool if you could list turrets, missiles, bombs, ammo, etc. as separate bulkhead profiles to make them easier to find if that make sense?

**Comment by Gedas-S** [comment]
You can try right-clicking the bda category button in the editor, and enabling subcategories. ;)

**Comment by Gedas-S** [comment]
Should be fixed in 1.3.1 (just out).
